src/extension.ts

```diff
@@ -66,14 +68,27 @@ interface LogChannel {
   hasMoreLogs?: boolean;
 }
 
+const loadEnvironments = async (
+  environmentsDataProvider: Env0EnvironmentsProvider,
+  environmentsTree: vscode.TreeView<Environment>
+) => {
+  try {
+    const { currentBranch } = getGitData();
+    environmentsTree.message = `loading environments from branch ${currentBranch}…`;
+  } catch {
```

Comment:
When opening visual studio code all extensions activated together, therefore git extension sometimes is not available.

Reply:
isn't there a way to verify/wait for a plugin to load? instead of catching errors

Reply:
Basically, most of the time the environments are loaded before the git extension is available.

Reply:
I can build something that will pull the branch name (i will try to use` vscode.extensions.onDidChange`) and update the loading message once we have it but i am not sure its worth it...

Reply:
can't you just simply stay in loading state until the dependency is ready?

I asked Bard, can you give it try? I think it should be simple.
```
const dependencyPlugin = vscode.extensions.getExtension('my-dependency-plugin');

function checkIfDependencyPluginLoaded() {
  if (dependencyPlugin.loaded) {
    // The dependency plugin is loaded.
    return;
  }
  }
```

Reply:
full answer (just for further reading)

Sure, here are the steps on how to wait for a dependency plugin to finish loading before running your VS Code extension:

1. Create a function that will check if the dependency plugin is loaded. This function can use the `vscode.extensions.getExtension` method to get the dependency plugin object. The `loaded` property of the dependency plugin object will be `true` if the plugin is loaded.
2. Create a timer that will call the check function every few milliseconds. The timer should continue to run until the check function returns `true`.
3. In your extension's `activate` method, call the timer.

Here is an example of how to do this:

```
const dependencyPlugin = vscode.extensions.getExtension('my-dependency-plugin');

function checkIfDependencyPluginLoaded() {
  if (dependencyPlugin.loaded) {
    // The dependency plugin is loaded.
    return;
  }

  // The dependency plugin is not loaded yet.
  setTimeout(checkIfDependencyPluginLoaded, 100);
}

const timer = setInterval(checkIfDependencyPluginLoaded, 100);

async function activate() {
  // Wait for the dependency plugin to load.
  await new Promise((resolve) => {
    timer.on('timeout', resolve);
  });

  // The dependency plugin is loaded. Do something.
}
```

Once the dependency plugin is loaded, the `checkIfDependencyPluginLoaded` function will return `true` and the timer will be stopped. Your extension's `activate` method will then be able to continue.

I hope this helps! Let me know if you have any other questions.

Reply:
030ac9f0ecfa08d35525563ad4a80d5094daf5a2
after adding vscode.git as extensionDependencies in package json, our extension is loaded after vscode.git extension loaded.
but still it takes some time until the git extension initialised and can return the current branch so i added retry